amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/activity/ActivityImporter.java

```diff
@@ -759,32 +655,6 @@ protected void updateRoleFundings() {
         }
     }
     
-    /**
-     * Updates activity org roles (missing org roles from fundings)
-     */
-    protected void updateOrgRoles() {
-        for (AmpFunding f : newActivity.getFunding()) {
-            if (f.getSourceRole() != null && f.getAmpDonorOrgId() != null) {
-                boolean found = false;
-                for (AmpOrgRole role : newActivity.getOrgrole()) {
-                        if (role.getRole().getRoleCode().equals(f.getSourceRole().getRoleCode()) 
-                                && role.getOrganisation().getAmpOrgId().equals(f.getAmpDonorOrgId().getAmpOrgId())) {
-                            found = true;
-                            break;
-                        }
-                }
-            
-                if (!found) {
-                    AmpOrgRole role = new AmpOrgRole();
-                    role.setOrganisation(f.getAmpDonorOrgId());
-                    role.setActivity(newActivity);
-                    role.setRole(f.getSourceRole());
-                    newActivity.getOrgrole().add(role);
```

Comment:
New org role generation doesn't seem to be covered through `InterchangeableBackReference`. Only existing org roles activity will be set.

Reply:
This code is creating any missing org roles that were specified in funding. However at the same time, we have a validator which does not allow activities with funding for which there is no corresponding org role defined. Thus this code is never invoked and unnecessary.

Reply:
Thanks for clarification, so when validator was added this code became obsolete - then I agree is safe to remove